web/packages/teleterm/src/ui/Search/pickers/ActionPicker.tsx

```diff
@@ -789,13 +795,22 @@ export function AppItem(props: SearchResultItem<SearchResultApp>) {
   );
 }
 
-function getAppItemCopy($appName: React.JSX.Element, app: tsh.App) {
+function getAppItemCopy(
+  $appName: React.JSX.Element,
+  app: tsh.App,
+  isVnetSupported: boolean
+) {
   if (app.samlApp) {
     return <>Log in to {$appName} in the browser</>;
   }
   if (isWebApp(app) || app.awsConsole) {
     return <>Launch {$appName} in the browser</>;
   }
+
+  // TCP app
+  if (isVnetSupported) {
+    return <>Connect with VNet to {$appName}</>;
```

Comment:
At first, I made it say "Connect through VNet to [app]". But "Connect through VNet" does not roll off the tongue at all.